modules/ossm-integrating-with-user-workload-monitoring.adoc

```diff
@@ -0,0 +1,176 @@
+= Integrating with user-workload monitoring
+
+By default, OpenShift Service Mesh (OSSM) installs the Service Mesh Control Plane (SMCP) with a dedicated instance of Prometheus for collecting metrics from a mesh. However, production systems need more advanced monitoring systems, like OpenShift Monitoring for user-defined projects.
+
+The following steps show how to integrate Service Mesh with User-Workload Monitoring.
+
+.Procedure
+
+. Enable monitoring in user-defined projects by adding the following configuration to the `cluster-monitoring-config` config map:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    enableUserWorkload: true
+----
+
+. Complete one of the following options to configure the Service Mesh control plane for User-Workload Monitoring.
+** Either disable the installation of the default Prometheus instance:
++
+[source,yaml]
+----
+apiVersion: maistra.io/v2
+kind: ServiceMeshControlPlane
+metadata:
+  name: basic
+  namespace: istio-system
+spec:
+  addons:
+    prometheus:
+      enabled:  false # <1>
+  meshConfig:
+    extensionProviders:
+    - name: prometheus # <2>
+      prometheus: {}
+  security:
+    manageNetworkPolicy: false # <3>
+----
+<1> Disable the default Prometheus instance provided by OSSM.
+<2> Configure the Prometheus extension provider.
+<3> Disable OSSM managed network policies. 
++
+[NOTE]
+====
+This option disables all of your OSSM managed network policies.
+====
++
+** Or add an additional network policy to enable ingress from `user-workload-monitoring`:
++
+[source,yaml]
+----
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: user-workload-access
+  namespace: bookinfo
+spec:
+  ingress:
+  - from:
+    - namespaceSelector:
+        matchLabels:
+          network.openshift.io/policy-group: monitoring
+  podSelector: {}
+  policyTypes:
+  - Ingress
+----
++
+[NOTE]
+====
+This option lets you keep your OSSM managed network policies.
+====
++
+
+. Apply a telemetry object to enable traffic metrics in Istio proxies.
++
+[source,yaml]
+----
+apiVersion: telemetry.istio.io/v1alpha1
+kind: Telemetry
+metadata:
+  name: enable-prometheus-metrics
+  namespace: istio-system
+spec:
+  metrics:
+  - providers:
+    - name: prometheus
+----
++
+[NOTE]
+====
+When a telemetry object is created in the control plane namespace, it applies to all workloads in a mesh. You can apply the telemetry object in selected namespaces or workloads by setting `spec.selector.matchLabels`. 
+====
+
+. Apply a ServiceMonitor to monitor the Istio control plane.
++
+[source,yaml]
+----
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: istiod-monitor
+  namespace: istio-system
+spec:
+  targetLabels:
+  - app
+  selector:
+    matchLabels:
+      istio: pilot
+  endpoints:
+  - port: http-monitoring
+    interval: 30s
```

Comment:
```suggestion
    interval: 30s
    relabelings:
    - action: replace
      replacement: "<SMCP_NAME>-<SMCP_NAMESPACE>" <2>
      targetLabel: mesh_id
```
@fjglira could you also add this change to our e2e test and then verify all in Kiali in OCP dashboard?

Reply:
@gwynnemonahan please apply this suggestion.